scanner/src/main/kotlin/CommandLinePathScannerWrapper.kt

```diff
@@ -36,4 +40,22 @@ abstract class CommandLinePathScannerWrapper(name: String) : PathScannerWrapper,
     abstract val configuration: String
 
     override val details by lazy { ScannerDetails(name, getVersion(), configuration) }
+
+    final override fun scanPath(path: File, context: ScanContext): ScanSummary {
+        val startTime = Instant.now()
+        val result = runScanner(path, context)
+        val endTime = Instant.now()
+        return createSummary(result, startTime, endTime)
+    }
+
+    /**
+     * Run the scanner on the given [path] with the given [context].
+     */
+    abstract fun runScanner(path: File, context: ScanContext): String
+
+    /**
+     * Create a [ScanSummary] from the scan [result] in a scanner-native format. If the [result] itself does not contain
+     * time information, [startTime] and [endTime] may be used instead.
+     */
+    abstract fun createSummary(result: String, startTime: Instant, endTime: Instant): ScanSummary
```

Comment:
What's the reason for limiting the support for scanners on clearly defined to just command line based scanners? e.g. extend the API of `CommandLinePathScanner`. 

When viewing the API from scanner perspective, it looks that now implementation details get exposed / it's getting a bit more complex. 

As already proposed in a comment for a following commit, we could alternatively either / or do:

1.  Extend ScannerFactor by a function `createClearlyDefinedResultParser(toolVersion: String): ClearlyDefinedResultParser?`
2. Create a separate plugin factory / interface with above function.

note: code can still be shared, so function implementing that interface would just be a thin wrapper around the real logic used by scanner as well as by storage.

Reply:
> I'd prefer to not put any ClearlyDefined-specific code into the scanner API.

In order to reply on this, I'd need clarification what you mean by scanner API? Do you mean `Scanner.Factory` (which could be considered only the loading meachanism of `Scanner`, in constrast to `Scanner` as scanner API. Please clarify.

In any case, my essence of my proposal was meant to have a complete separate interface for the parsing of clearly defined results. So, the proposal is the exact opposite of mixing things. It'd be mixing things way less IIUC compared to the proposed code, wouldn't it?

Reply:
> In order to reply on this, I'd need clarification what you mean by scanner API?

Any code in the "scanner" Gradle module that would remain there after [moving scan storages to separate plugin projects](https://github.com/oss-review-toolkit/ort/issues/6603). So yes, that would include `Scanner.Factory`.

> In any case, my essence of my proposal was meant (I fear it's not clear enough still) to have a complete separate interface for the parsing of clearly defined results.

As [mentioned](https://github.com/oss-review-toolkit/ort/pull/7242#discussion_r1254185385), I like the idea of your `ClearlyDefineResultParser` interface. But any related code should not "Extend ScannerFactor(y)" (sic), but go into code that is specific to (a) ClearlyDefined (scan storage plugin).

Reply:
> But any related code should not "Extend ScannerFactor(y)" (sic), but go into code that is specific to (a) ClearlyDefined (scan storage plugin).

Sure, that would work for me nicely as well. (A minor detail could be where that interface implementing classes would reside. My preference would be under e.g. `plugins/scanners/scancode` or `plugins/scanners/licensee`, but I would tolerate other locations as well, because I'm already fine as soon as the class hierarchy / interfaces are fixed.

So, if we plan to do that, why should be then now (in this PR) change the interface of the path scanner wrapper. 
(I'm not saying it as complete objection against, but want to learn the arguments why this could make sense, because I fear I do not yet fully get what you have in mind)

Reply:
> So, if we plan to do that, why should be then now (in this PR) change the interface of the path scanner wrapper.

I'm not completely understanding the sentence, but I believe you're asking "why not doing it now". And the simple answer is that I was never planning to do refactoring in the first place, but it turned out to be a prerequisite for being able to move the ScanCode scanner wrapper to a separate plugin project, which is the actual topic I'm working on. As I have the refactoring for that already done, and I just need to rebase it onto this PR, I'd like to get this merged ASAP to be able to finish one topic before starting the other.